apps/omg_eth/lib/omg_eth/root_chain.ex

```diff
@@ -21,65 +21,52 @@ defmodule OMG.Eth.RootChain do
   Should remain simple and not contain any business logic, except being aware of the RootChain contract(s) APIs.
   """
 
-  alias OMG.Eth
-  alias OMG.Eth.Config
-
   require Logger
   import OMG.Eth.Encoding, only: [to_hex: 1, from_hex: 1, int_from_hex: 1]
 
-  @type optional_address_t() :: %{atom => Eth.address()} | %{atom => nil}
-  @type in_flight_exit_piggybacked_event() :: %{
-          owner: <<_::160>>,
-          tx_hash: <<_::256>>,
-          output_index: non_neg_integer
-        }
+  alias OMG.Eth
+  alias OMG.Eth.Configuration
+  alias OMG.Eth.RootChain.Abi
+  alias OMG.Eth.RootChain.Rpc
 
-  ########################
-  # READING THE CONTRACT #
-  ########################
+  @type optional_address_t() :: %{atom => Eth.address()} | %{atom => nil}
 
-  # some constant-like getters to start
+  def get_mined_child_block() do
+    child_block_interval = Configuration.child_block_interval()
+    mined_num = next_child_block()
+    mined_num - child_block_interval
+  end
 
-  @spec get_child_block_interval() :: {:ok, pos_integer()} | :error
-  def get_child_block_interval(), do: Application.fetch_env(:omg_eth, :child_block_interval)
+  def next_child_block() do
+    contract_address = Configuration.contracts().plasma_framework
+    %{"block_number" => mined_num} = get_external_data(contract_address, "nextChildBlock()", [])
+    mined_num
+  end
 
-  @doc """
-  This is what the contract understands as the address of native Ether token
-  """
-  @spec eth_pseudo_address() :: <<_::160>>
-  def eth_pseudo_address(), do: Eth.zero_address()
+  def blocks(mined_num) do
```

Comment:
hm hm, the reason I kept this naming is because `blocks/1` and `next_child_block/0` are direct calls to plasma contracts, and return the result "unmodified".
where as `get_mined_child_block /0` is calculated as a diff between the next child block and the leap (interval).
so... I was like, these are not the same in principal maybe I shouldn't name them the same. IDK... should I change?

Reply:
>  these are not the same in principal maybe I shouldn't name them the same

I'm kinda inclined towards that being the underlying implementation that the function user doesn't need to know. My first reaction to this was "is this function changing something?", the `get_*` just makes it clear it's a getter and aligns with other module APIs e.g. Map, Ecto, Keyword, etc.

Subjective though, might just need a 3rd opinion @achiurizo

Reply:
I think I started with `get_*` names initially but I found it harder to keep track of the different namings we had between elixir and the contracts. I'm leaning towards we match elixir namings to the contract function names(same with the variable names), so I'm for `def blocks`.

That means we'll need to think about some sort of standard for other functions around it, like the `get_mined_*` being calculated.  I think `get_*` could work well.